components/org.wso2.carbon.identity.api.server.input.validation/org.wso2.carbon.identity.api.server.input.validation.v1/src/main/java/org/wso2/carbon/identity/api/server/input/validation/v1/core/ValidationRulesManagementApiService.java

```diff
@@ -0,0 +1,415 @@
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_VALIDATION_PARAM_NOT_SUPPORTED;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_VALIDATOR_NOT_SUPPORTED;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.SUPPORTED_PARAMS;
+
+/**
+ * Calls internal osgi services to perform input validation management related operations.
+ */
+public class ValidationRulesManagementApiService {
+
+    private static final Log log = LogFactory.getLog(ValidationRulesManagementApiService.class);
+
+    /**
+     * Method to get input validation configuration.
+     *
+     * @param tenantDomain  Tenant Domain.
+     * @return ValidationConfigModal.
+     */
+    public List<ValidationConfigModel> getValidationConfiguration(String tenantDomain) {
+
+        try {
+            List<ValidationConfiguration> configurations = InputValidationServiceHolder.getInputValidationMgtService()
+                    .getInputValidationConfiguration(tenantDomain);
+            return buildResponse(configurations);
+        } catch (InputValidationMgtException e) {
+            if (ERROR_CODE_INPUT_VALIDATION_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Can not find a validation configurations for tenant: " +
+                            tenantDomain, e);
+                }
+            }
+            throw handleInputValidationMgtException(e, ERROR_CODE_ERROR_GETTING_VALIDATION_CONFIG, tenantDomain);
+        }
+    }
+
+    /**
+     * Method to update input validation configuration.
+     *
+     * @param validationConfigModel Validation Configuration Model.
+     * @param tenantDomain          Tenant domain name.
+     */
+    public List<ValidationConfigModel> updateInputValidationConfiguration(
+            List<ValidationConfigModel> validationConfigModel, String tenantDomain) {
+
+        try {
+            List<ValidationConfiguration> requestDTO = buildRequestDTOFromValidationRequest(validationConfigModel);
+            validateProperties(requestDTO, tenantDomain);
+            List<ValidationConfiguration> configurations = InputValidationServiceHolder.getInputValidationMgtService()
+                    .updateInputValidationConfiguration(requestDTO, tenantDomain);
+            return buildResponse(configurations);
+        } catch (InputValidationMgtException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Unable to update validation configuration for tenant: " + tenantDomain, e);
+            }
+            throw handleInputValidationMgtException(e, ERROR_CODE_ERROR_UPDATING_VALIDATION_CONFIG, tenantDomain);
+        }
+    }
+
+    /**
+     * Method to validate input values.
+     *
+     * @param tenantDomain      Tenant domain.
```

Comment:
Formatting issue. Additional spaces between the parameter and the description. Check other places too.